src/cts/src/TechChar.cpp

```diff
@@ -551,28 +536,32 @@ void TechChar::initCharacterization()
 
   // Defines the different wirelengths to test and the characterization unit.
   unsigned wirelengthIterations = options_->getCharWirelengthIterations();
-  unsigned maxWirelength = (charBuf_->getHeight() * 10)
-                           * wirelengthIterations;  // Hard-coded limit
-  if (options_->getWireSegmentUnit() == 0) {
-    unsigned charaunit = charBuf_->getHeight() * 10;
-    options_->setWireSegmentUnit(charaunit);
+  if (options_->getWireSegmentUnitDbu() == 0) {
+    unsigned charaunitDbu = charBuf_->getHeight() * 10;
+    options_->setWireSegmentUnitDbu(charaunitDbu);
   } else {
-    // Updates the units to DBU.
-    unsigned segmentDistance = options_->getWireSegmentUnit();
-    options_->setWireSegmentUnit(segmentDistance * dbUnitsPerMicron);
+    // user has passed the segment unit in Micron, so we need to convert
+    // it to DBU to keep correctness and uniformity.
+    options_->setWireSegmentUnitDbu(options_->getWireSegmentUnitDbu()
+                                    * dbUnitsPerMicron);
```

Comment:
The conversion from microns to dbu should be done on the way in from tcl rather the delayed to here.  I want to remove microns as much as possible from the core code.